miden-lib/build.rs

```diff
@@ -1,14 +1,66 @@
 use assembly::{LibraryNamespace, MaslLibrary, Version};
+use std::env;
+use std::fs;
 use std::io;
+use std::path::Path;
+use std::{fs::File, io::BufRead, io::BufReader, io::Write};
 
 // CONSTANTS
 // ================================================================================================
-const ASM_DIR_PATH: &str = "./asm";
 const ASL_DIR_PATH: &str = "./assets";
```

Comment:
Is there a reason to remove `ASM_DIR_PATH` constant? I think it would be useful to have it as the root directory of the library might change in the future, and it would be good to have just one place to update it.

Reply:
added a const name `ASM_DIR`, removed the `_PATH` since it is only the folder name and does not contain `/` in it.

Reply:
I think we might want to use a path here rather than just a directory name. For example, in https://github.com/0xPolygonMiden/miden-base/issues/246 we are proposing to move the library under `/asm/miden` and add a separate directory for note scripts under `/asm/scripts`. We can, of course, do it as a part of #246 work, but if it doesn't create too much work, might be good to fix as a part of this PR.

Reply:
added `_PATH` back